x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/AutoFollowStats.java

```diff
@@ -117,11 +117,15 @@ public AutoFollowStats(StreamInput in) throws IOException {
         numberOfFailedRemoteClusterStateRequests = in.readVLong();
         numberOfSuccessfulFollowIndices = in.readVLong();
         if (in.getVersion().onOrAfter(Version.V_6_7_0)) {
-            recentAutoFollowErrors = new TreeMap<>(in.readMap(StreamInput::readString,
-                in1 -> new Tuple<>(in1.readZLong(), in1.readException())));
+            // need some explicit type casting here to infer arguments correctly in Eclipse IDE
+            Map<String, Tuple<Long, ElasticsearchException>> autoFollowErrors = in.readMap(StreamInput::readString,
```

Comment:
It seems nice to do this in a separate commit. Totally up to you though, I don't want to make a big hassle.

Reply:
Its already in another PR, this will go away when merging in master, I just needed it to work on this branch without compile errors from my IDE.

Reply:
Merging in 6.x I mean. Its 3863435a0da1824013214642971041fe8317883a